### PR TITLE
Update the toxicity metric [en, ja]

### PR DESCRIPTION
**Pull request by conan1024hao**
# New Toxicity vs Old Toxicity experiment
## Data
- 150 samples from the Rakuten Japanese Toxicity Dataset ([README](https://rak.app.box.com/s/khubss0hh5iyh45pa2n1p3yzbz1dswyc) [Dataset](https://rak.app.box.com/s/p0jc5nnz2hc1h71zsjsm7p7fbpj0bsl5))
  - 50 `Toxic` samples, 50 `Has Toxic Expression` samples, 50 `Nontoxic` samples.

## Metric
- `AzureOpenAIEvalClient` (`gpt-35-turbo-16k`, `gpt-4-turbo`)
  - Some samples in the Rakuten dataset are quite long.
- Old prompt: https://github.com/citadel-ai/langcheck/blob/main/src/langcheck/metrics/prompts/en/metrics/toxicity.j2
- New prompt: https://github.com/citadel-ai/langcheck/blob/update-toxicity-metric/src/langcheck/metrics/prompts/en/metrics/toxicity.j2
  -  @liwii Please take a look at [the Japanese version prompt](https://github.com/citadel-ai/langcheck/blob/update-toxicity-metric/src/langcheck/metrics/prompts/ja/metrics/toxicity.j2).

## Evaluation
- `exact_match` was used as the evaluation metric.
- If the model outputs `None` as the score, we consider this sample incorrect.
- For the model using the old prompt, we transform the scores `0` to the label `Nontoxic`, `0.25, 0.5` to the label `Has Toxic Expression`, `0.75, 1.0` to the label `Toxic`.

## Results
- There is no significant difference in `exact_match`. 
  
  | Model | old prompt | new prompt |
  |--|--|--|
  | GPT3.5 Turbo | 0.4133 (62 / 150) | 0.3867 (58 / 150) | 
  | GPT4 Turbo | 0.54 (81 / 150) | 0.5267 (79 / 150)| 

- Label distribution
  - All models' label distributions are quite shifted. This might be because the data in the Rakuten dataset is long, making the 'toxic' part only a small portion.
  - The definition for toxicity such as `obscene, discriminatory, violent` didn't help too much.

  | Model | Prompt | Nontoxic | Has Toxic Expression | Toxic | None |
    |--|--|--|--|--|--|
    |GPT3.5 Turbo | old prompt |60 | 44 | 12 | 34 |
    |GPT3.5 Turbo  | new prompt | 82 | 34 | 10 | 24 | 
    |GPT4 Turbo | old prompt | 55 | 60 | 14| 21 |
    |GPT4 Turbo  | new prompt | 59 | 63 | 3 | 25 |

### CODE REVIEW

**Comment by conan1024hao** [comment]
# Experiment v2
## Data
- English: 140 samples from [the Jigsaw Multilingual Toxic Comment dataset](https://www.kaggle.com/competitions/jigsaw-multilingual-toxic-comment-classification/data?select=jigsaw-toxic-comment-train.csv)
  - 70 toxic samples, 70 nontoxic samples
- Japanese: 140 samples from [the LINE toxic dataset ](https://github.com/inspection-ai/japanese-toxic-dataset)
  - Samples are considered toxic if they receive at least one vote for being toxic or very toxic, while samples with all votes as nontoxic are considered nontoxic.
  - 70 toxic samples, 70 nontoxic samples

## Metric
- `OpenAIEvalClient` (`gpt-3.5-turbo`, `gpt-4-turbo`)
  - By using the normal OpenAI client, `None` is no longer outputted
- Update the prompt into [a binary one](https://github.com/citadel-ai/langcheck/blob/752c84d4f081f58872834c38c2ab629c58d88eb6/src/langcheck/metrics/prompts/en/metrics/toxicity.j2).

## Results
| Model | EN | JA | All|
|--|--|--|--|
|GPT3.5 Turbo |0.8929 | 0.5714| 0.7321|
| GPT4 Turbo |0.8643 | 0.6786| 0.7714|

## Label Distribution
| Model | Language | Nontoxic | Toxic |
|--|--|--|--|
|GPT3.5 Turbo | EN |75 | 65|
|GPT3.5 Turbo | JA | 128 | 12|
| GPT4 Turbo | EN | 79 | 61 | 
| GPT4 Turbo | JA | 111 | 29|

File: src/langcheck/metrics/prompts/en/metrics/toxicity.j2 [hunk 1]
```diff
@@ -1,9 +1,19 @@
+{#
+Ref:
+    Our prompt was created with reference to the Rakuten Japanese Toxicity Dataset dataset
+    https://rak.app.box.com/s/khubss0hh5iyh45pa2n1p3yzbz1dswyc
+#}
```
Comment (yosukehigashi) [inline src/langcheck/metrics/prompts/en/metrics/toxicity.j2:5]: I think we can remove this now?

**Review by yosukehigashi — APPROVED** [review]
LGTM!! Just one comment

**Comment by yosukehigashi** [comment]
@liwii any idea what's going on with the ruff format check here? It was working fine before, and it also works for me when I run it locally
image

**Comment by yosukehigashi** [comment]
looks like there are some breaking changes due to https://github.com/astral-sh/ruff/releases/tag/0.6.0

I'll just pin it to <0.6 for now and deal with it in another PR

**Pull request merged**